### PR TITLE
Use $strict param in functions that have it

### PR DESCRIPTION
**Pull request by carusogabriel**
Some functions, e.g. `in_array`, have a third param to enable strict comparison :shield:

### CODE REVIEW

**Review by lcobucci — CHANGES_REQUESTED** [review]
IMO we should also use https://github.com/phpstan/phpstan-strict-rules to enforce this

**Comment by carusogabriel** [comment]
I used https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.9/src/Fixer/Strict/StrictParamFixer.php, but if you want, I can change the PR :sweat_smile:

**Comment by lcobucci** [comment]
@carusogabriel of course we can solve this using various tools, but the point is that IMO doing the strict checks from phpstan is preferable to prevent bugs in the future

**Comment by carusogabriel** [comment]
@lcobucci I'm just installing PHP 7.2 locally, I'll modify this PR to use `phpstan-strict-rules` :smile:

**Comment by Majkl578** [comment]
>  I'll modify this PR to use phpstan-strict-rules

I'm not sure whether it's wise at this moment. It will cause a lot of errors imho, especially implicit bool casts in conditions. :/ Lets postpone it for now and just switch on these `$strict` params, there could also be a PHPCS rule for that (requiring explicit value for `$strict` params).

**Comment by Majkl578** [comment]
This will need a rebase now, sorry. :/

**Comment by carusogabriel** [comment]
No problem, I’ll rebase and solve the conflicts

**Review by Majkl578 — APPROVED** [review]
Don't see any apparent coercion issue, so LGTM.

**Comment by Majkl578** [comment]
@lcobucci @carusogabriel Let's address `phpstan-strict-rules` later, PHPStan build is currently red anyway and needs fixes first.

File: tests/Doctrine/Tests/ORM/Functional/Ticket/DDC3192Test.php [hunk 1]
```diff
@@ -155,7 +155,7 @@ public function convertToDatabaseValue($value, AbstractPlatform $platform)
      */
     public function convertToPHPValue($value, AbstractPlatform $platform)
     {
-        return array_search($value, self::$map);
+        return array_search($value, self::$map, true);
```
Comment (Majkl578) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/DDC3192Test.php]: This change has actually broken the test:
```
1) Doctrine\Tests\ORM\Functional\Ticket\DDC3192Test::testIssue
Failed asserting that false matches expected 'BYR'.

/www/doctrine/doctrine2/tests/Doctrine/Tests/ORM/Functional/Ticket/DDC3192Test.php:55
```

There is likely string to integer coercion that doesn't work after enabling strict mode.

Comment (carusogabriel) [reply tests/Doctrine/Tests/ORM/Functional/Ticket/DDC3192Test.php]: Have you reverted the change or should I? Sorry about that!

Comment (Majkl578) [reply tests/Doctrine/Tests/ORM/Functional/Ticket/DDC3192Test.php]: Fixed it in d791f7f6ca77ab9acb282317664d1ea7316bae8d.

**Pull request merged**